core/src/locktower.rs

```diff
@@ -113,8 +118,11 @@ impl Locktower {
             if lamports == 0 {
                 continue;
             }
-            let mut vote_state: VoteState = VoteState::deserialize(&account.data)
-                .expect("bank should always have valid VoteState data");
+            let vote_state = VoteState::from(&account);
+            if vote_state.is_none() {
+                continue;
```

Comment:
Maybe a error print here? We what used to be fatal is now just ignored silently? Unlikely this should happen but still.

Reply:
it could happen.  vote accounts are not necessarily under the control of the vote program.  you can create an account with no data and give it to the vote program.  in such a case, it won't be counted in this staking simulation

crashing this node because someone put a corrupt vote account in the bank would be wrong

Reply:
Yeah crashing isn't helpful but maybe a log and data point?

Reply:
https://github.com/solana-labs/solana/pull/4424